src/main/java/com/answerdigital/benhession/academy/answerkingweek2/controllers/OrderController.java

```diff
@@ -32,19 +32,25 @@ public OrderController(final OrderService orderService) {
     }
 
     @PostMapping
-    public ResponseEntity<Order> addOrder(@Valid @RequestBody final AddOrderRequest addOrderRequest) {
-        return new ResponseEntity<>(orderService.addOrder(addOrderRequest.address()), HttpStatus.CREATED);
+    public ResponseEntity<Order> addOrder(@RequestBody final OrderRequest orderRequest) {
+        return new ResponseEntity<>(orderService.addOrder(orderRequest), HttpStatus.CREATED);
     }
 
     @GetMapping(path = "/{orderId}")
     public ResponseEntity<Order> getOrder(@PathVariable @NotNull final Long orderId) {
         return ResponseEntity.ok(orderService.findById(orderId));
     }
 
+    @PutMapping("/{orderId}")
+    public ResponseEntity<Order> updateOrder(@PathVariable @NotNull final Long orderId,
+                                             @RequestBody final OrderRequest orderRequest) {
```

Comment:
same as above, @Valid annotation should be added for OrderRequest Body

Reply:
similar to my previous comment, @Valid is not a class-level annotation.

Reply:
Agreed, using @Valid at class-level seems to be unstable. Will revert all controllers back to parameter-level validation.

Reply:
Fixed